rfcs/greedy-stageless.md

```diff
@@ -0,0 +1,233 @@
+  - _Removed_ ~~`.at_start()`~~
+  - _Removed_ ~~`.at end()`~~
+
+### Scheduling when to apply commands
+
+Systems with commands have two parts, a parallel part that can be run in parallel with other systems that the user codes and a command queue that needs to be applied to the world during a hard sync point. Since these happen at different times they should be scheduled separately. This is done through three new system descriptor builder functions.
+
+- `.commands_label(Label)` gives the point where the command queue is applied a `Label` which other systems can be scheduled around.
+- `.commands_after(Label)` is used to schedule the command queue to be applied after `Label`
+- `.commands_before(Label)` is used to schedule the command queue to be applied before `Label`
+
+```rs
+fn main() {
+    App::new()
+        // add a label to systemA's commands
+        .add_system(systemA.label("System A").commands_label("System A Commands"))
+
+        // places systemB after systemA's commands are applied
+        .add_system(systemB.after("System A Commands"));
+
+        // schedule systemC between systemA's commands are applied
+        .add_system(systemC.after("System A").before("System A Commands"));
+
+        // schedule systemD's commands after system A's commands
+        .add_system(systemD.commands_after("System A Commands"))
+
+        // schedule systemE's commands before system A's commands
+        .add_system(systemE.commands_before("System A commands"));
+}
+```
+
+### Exclusive Systems
+
+Exclusive systems are systems that depend on direct world access. They are marked with the `.exclusive()` builder function and only have the `World` parameter. The `label`, `before`, and `after` apis can be used to order exclusive systems and can be mixed with labels for parallel systems.
+
+### Run Criteria
+
+Run criteria determine whether or not a system will run. They now are separated from LoopingCriteria. A RunCriteria system is a system that returns a bool with true for run the dependent system and false for not running the system. Systems without run criteria will be run once per tick.
+
+```rs
+fn should_run_once(hasRun: Local<bool>) -> bool {
+    if *hasRun {
+        false
+    } else {
+        *hasRun = true
+        true
+    }
+}
+
+fn spawn_player(mut commands: Commands) {
+    commands.spawn().insert(Player);
+}
+
+fn main() {
+    App::new()
+        .add_system(
+            spawn_player.with_run_criteria(should_run_once)
+        );
+}
+```
+
+### Looping Criteria
+
+A looping criteria is a system that returns a bool. Returning true makes a system reevaluate its run criteria and false means the system is done running for this tick. Adding looping criteria introduces an implicit requirement for a hard sync. The hard sync is required to reevaluate looping criteria and run criteria. Systems scheduled after a looping system will run after the looping criteria returns false.
+
+```rs
+fn loop_twice(loop_count: Local<int32>) {
```

Comment:
`i32`, not `int32`. Maybe `usize` is preferable in this case.

You also missed `-> bool`